stellargraph/core/graph.py

```diff
@@ -832,6 +833,13 @@ def _unique_type_triples(self, *, return_counts, selector=slice(None)):
 
         return zip(*unique_ets)
 
+    def _edge_metrics_by_type_triple(self, *, selector=slice(None), metrics):
+        src_ty, rel_ty, tgt_ty = self._edge_type_triples(selector)
+        df = pd.DataFrame(
+            self._edges.weights, index=[src_ty, rel_ty, tgt_ty], columns=["weight"]
```

Comment:
If this supports a `selector` argument, I think it needs to be applied here too:

```suggestion
            self._edges.weights[selector], index=[src_ty, rel_ty, tgt_ty], columns=["weight"]
```

or else the length of this will be different to the length of the index.

Alternatively, we could not support `selector` at all and just call `src_ty, rel_ty, tgt_ty = self._edge_type_triples()`.

Reply:
I might just remove `selector`, rather than make `_edge_metrics_by_type_triple` more flexible than it needs to be when it's only being used in the `info` method for this specific purpose.